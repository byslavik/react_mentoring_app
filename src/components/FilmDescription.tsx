import * as React from 'react';

import { connect } from 'react-redux';

import {getFilmsByQuery} from '../actions/films';

import {urlParams} from './../models/common.models'; 

import * as s from './scss/FilmDescripiton.scss';

import {FullFilm, Film} from '../models/film.model';
import {ReduxStore} from './../models/redux.model';

import {SETTINGS} from './../settings';

interface FilmProps {
  currentFilm?: FullFilm;
  getFilmsByQuery?: any;
  match?:any
}

export class FilmDescription extends React.PureComponent <FilmProps, any> {
    componentWillMount() {
      this.getItems(this.props.match.params.id)
    }
    componentWillReceiveProps(nextProps:any){
      if(this.props.match.params.id != nextProps.match.params.id) {
        this.getItems(nextProps.match.params.id)
      }
    }
    getItems(id:string) {
      let params:urlParams = {
        type: 'movie',
        query: id
      }
      this.props.getFilmsByQuery(params);
    }
    render() {

      const filmData = this.props.currentFilm;
      
        return (<article className={s.filmContainer} key={this.props.match.params.id}>
          <div className={s.imgWrap}>
            <img src={SETTINGS.imgUrl + filmData.poster_path} alt={filmData.original_title}/>
          </div>
          <div className={s.filmDescription}>
            <h1>{filmData.title} <span>{(Math.round(filmData.popularity * 10)/10)}</span></h1>
            <ul className={s.fimInfo}>
              <li>{filmData.release_date}</li>
            </ul>
            <p className={s.filmSummary}>{filmData.overview}</p>
          </div>
        </article>)
    }
}

export default connect<any, any, any>(
  (state: ReduxStore) => ({
    currentFilm: state.films.currentFilm
  }),
  (dispatch: any) => ({
    getFilmsByQuery: (query:urlParams)=> {
      dispatch({type: 'GET_FILM', payload: {dispatchType: 'GET_FILM', urlParams: query}});
    }
   })
)(FilmDescription);