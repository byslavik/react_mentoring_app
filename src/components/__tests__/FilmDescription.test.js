import * as React from 'react';
import {FilmDescription} from '../FilmDescription';
import * as renderer from 'react-test-renderer';
import {filmData, match} from './testData'


describe('FilmDescription component', ()=>{
  it('FilmDescription component shows correctly', () => {
    const tree = renderer.create(
      <FilmDescription currentFilm={filmData} match={match} getFilmsByQuery={jest.fn()}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})