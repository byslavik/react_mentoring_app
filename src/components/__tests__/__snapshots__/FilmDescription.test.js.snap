// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDescription component FilmDescription component shows correctly 1`] = `
<article
  className={undefined}
>
  <div
    className={undefined}
  >
    <img
      alt="Original Title"
      src="https://image.tmdb.org/t/p/w500//somepath.jpg"
    />
  </div>
  <div
    className={undefined}
  >
    <h1>
      Title
       
      <span>
        100500
      </span>
    </h1>
    <ul
      className={undefined}
    >
      <li />
    </ul>
    <p
      className={undefined}
    >
      Some description
    </p>
  </div>
</article>
`;
